docs/optimization/feature_selection.md

```diff
@@ -0,0 +1,421 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.4
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+```{code-cell} ipython3
+:tags: []
+
+%load_ext autoreload
+%autoreload 2
+```
+
+# Feature Selection
+
+In a dataset, not every feature might be useful. Some could be noise, and some could show a strong relationship with the target variable. Often, using all the features decreases a model's performance and increases training time. Therefore, it's best to select the best subset of features that are relevant to the target variable before we train a model. This process is called feature selection. 
```

Comment:
`best to select the best subset` this is sounding redundant.
Briefly mention what are features vs target variables. mention overfitting here.